### PR TITLE
Distributing collectd modules

### PR DESCRIPTION
**Pull request by genuss**
Recently I pushed #35 , which added python module support. One little configuration variable there is ``collectd_settings.plugins.python.ModulePath``, which I don't like much. This variable will be almost everytime the same, however must be copied among pillar data.
Moreover, there is ``perl`` plugin, which alows perl execution, which has ``IncludeDir`` option. This dir may be equal to ``ModulePath`` from python module, if we put all scripts in one folder, which is convenient, when managing small amount of them.
I thought about this question and made: [this](https://github.com/saltstack-formulas/collectd-formula/compare/master...genuss:modules-distribution) (changing fqdn to nodename isn't important here). This is fairly simple.

Another question I want to arise is how one can distribute modules to minions? For my company I created a state, called ``collectd.modules``, which uses variables from this formula. It looks like that:
```
{% from "collectd/map.jinja" import collectd_settings with context %}

include:
  - collectd

{{ collectd_settings.moduledirconfig }}:
  file.recurse:
    - source: salt://collectd/modules/files
    - user: {{ collectd_settings.user }}
    - group: {{ collectd_settings.group }}
    - file_mode: 644
    - dir_mode: 755
    - watch_in:
      - service: collectd-service
```
It's ok for me, but for community this patch won't work, because everyone should be able to choose modules to add (to ``files`` folder). If I understand correctly, there is no way now to add files as pillar data (am I right?).

In short, two questions:
* Should we merge all module dirs in one variable?
* How do you manage files on minions and is it possible to make a universal solution here?

### CODE REVIEW

**Comment by genuss** [comment]
Made a PR. What do you think?

Another small question. I found some inaccuracies in repo and would like to correct them. Will you prefer many small PRs, or just one?

**Comment by puneetk** [comment]
just one please, and feel free to log all changes in the pull requests description.

**Comment by genuss** [comment]
Guys, what do you think about this one?

**Comment by 0xf10e** [comment]
Looks reasonable to me.

**Comment by babilen** [comment]
I like it, but would make `salt://collectd/modules/files` configurable, so that users can place modules whereever they want if they so desire.

**Comment by genuss** [comment]
I can surely make it configurable, but what for?
I thought ``salt://collectd/modules/files`` is pretty standard place to put files for state named ``collectd.modules``.

**Comment by babilen** [comment]
You are right, sorry for the noise and thank you!
